packages/nitro/types/shims.d.ts

```diff
@@ -9,3 +9,11 @@ declare module '#assets' {
   export function statAsset (id: string): Promise<AssetMeta>
   export function getKeys() : Promise<string[]>
 }
+
+declare module '#config' {
+  import type { PublicRuntimeConfig, PrivateRuntimeConfig, RuntimeConfig } from '@nuxt/kit'
+  export const privateConfig: PrivateRuntimeConfig
+  export const publicConfig: PublicRuntimeConfig
+  const runtimeConfig: RuntimeConfig
```

Comment:
@danielroe shouldn't this then be 
```suggestion
  const runtimeConfig: PublicRuntimeConfig & PrivateRuntimeConfig
```
since as you say above `RuntimeConfig` has the private config as optional (which would lead to unnecessary null checks server-side if you use `#config` there.

Reply:
Indeed you are right :+1

Reply:
[`ce90b23` (#1783)](https://github.com/nuxt/framework/pull/1783/commits/ce90b2386812c2bd966bc3083f9985e764f6cbb9)